Enhancement/UI enhancement

KvRae opened the pull request:
#  Feat: UI/UX Enhancements and Project Modernization

This pull request introduces a series of significant UI/UX improvements and modernizes the project's architecture and dependency management.

---

##  Key Changes

### 1. Dark Theme Support & Edge-to-Edge Display
- Implemented full support for a dark theme, which respects the system's setting. This enhances user comfort in low-light environments.
- Integrated the **Edge-to-Edge** feature to create an immersive, modern user experience.
- Status bar icons and text now correctly adapt their color in both light and dark modes, ensuring proper visibility.

### 2. New Splash Screen API
- Added a new splash screen using the `core-splashscreen` API.
- This provides a seamless and branded entry point into the application, replacing the default system splash screen.

### 3. UI Refinements & Code Cleanup
- **Support Screen**: Improved the top app bar navigation button in `SupportScreen.kt` for better consistency and interaction.
- **Code Cleanup**: Moved hardcoded strings into `strings.xml` to enhance maintainability and support future localization.

### 4. Gradle Version Catalog
- Migrated all Gradle dependencies from hardcoded strings in `build.gradle` files to a centralized **Version Catalog** (`libs.versions.toml`).
- This modernizes the build process, reduces redundancy, and simplifies dependency management.

### 5. Documentation
- Updated `README.md` to include badges for project status, license, and build state.

---

KvRae committed:
Update README.md

KvRae committed:
Feat: Ui Enhancement:
- Added splash screen to the app using the new splashscreen api
- Added dark theme support
- Enhanced SupportScreen.kt topbar navigation button
- Worked on code clean ups (moved hardcoded strings to strings.xml)
- replaced gradle dependencies imports using the new version catalog style
- fixed the items color on status bar text on light mode using the edge to edge feature

KvRae committed:
- Removed the AppViewModelTest.kt implementation since its not working yet

AmmarYasserAllaithy commented on app/src/main/res/values/strings.xml:
Although we promote Arabic lang, we should respect the localization. So please, these Arabic strings should translated to English.

And in the code we will enforce the ar locale.

KvRae replied:
Sure, as I mentioned, there are tons of hardcoded strings in the codebase, so it’s a bit hard to fix them all at once. I’ve been tackling them package by package, and by the next PR, I think it should all be done. inshallah

AmmarYasserAllaithy replied:
Good work, TYT 
ww are waiting for the changes, please when finish push the commit to this PR to merge them all.

KvRae replied:
Made a whole UI string resources rebase + forcing ar by default

AmmarYasserAllaithy commented on app/src/main/java/com/mafazaa/ainaa/ui/dialog/PermissionDialog.kt:
```diff
@@ -62,11 +70,21 @@ fun PermissionDialog(
                     TextButton(onClick = onDismiss) {
-                        Text("إلغاء")
+                        Text(
+                            text = "إلغاء",
```
You did well, continue what you started and use string resources

KvRae replied:
Yep, I already worked on the others, and it will be in the next pr inshallah, since the whole codebase has hardcoded strings

AmmarYasserAllaithy replied:
Well done, keep forward

AmmarYasserAllaithy reviewed (changes requested):
You did well, but we need some slight changes.
Please update them guided by the comments that I made.

KhaledHawwas reviewed (dismissed):
amazing work

KvRae committed:
refactor: use string resources for localization

KvRae committed:
feat: force Arabic locale and add backup rules

- Add `locales_config.xml` and `data_extraction_rules.xml`.
- Implement `LocaleHelper` to force the app's language to Arabic.
- Update `AppActivity` to use the new locale helper.
- Correct a constant name from `vpnAddress` to `VPN_ADDRESS`.

KvRae committed:
style(support): add padding to support screen

KvRae committed:
Merge branch 'dev' into enhancement/ui-enhancement

KvRae committed:
feat: internationalize EnableProtectionDialog

- Replaced hardcoded strings in `EnableProtectionDialog.kt` with string resources.
- Added English and Arabic translations for the new strings.

KvRae committed:
refactor: implement dark theme and RTL support

- Update theme colors for better dark mode appearance.
- Apply MaterialTheme colors and typography across various UI components (`AppBlockDialog`, `EnableProtectionDialog`, `LockScreenOverlay`, `ScreenshotOverlay`).
- Change the app name in `strings.xml` to Arabic script.
- Force Right-to-Left (RTL) layout and Arabic locale for the `LockScreenOverlay`.

AmmarYasserAllaithy commented on app/src/main/java/com/mafazaa/ainaa/domain/models/BlockReason.kt:
```diff
@@ -2,7 +2,7 @@ package com.mafazaa.ainaa.domain.models
-    data class UsingBlockedApp(val packageName: String) : BlockReason() {
+    data class UsingBlockedApp(val packageName: String ) : BlockReason() {
         override fun getName(): String {
             return "تطبيق محظور"
```
Can you please use string resources here!

AmmarYasserAllaithy reviewed (approved):
Well done.

*Just one change but recovered later*
